paywall/src/utils/polling.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * poll at a regular interval, with the ability to pause polling or end it
+ */
+export default function pollWithConditions(
+  pollingCallback,
+  intervalMs,
+  disablePolling,
+  handleError = () => {}
+) {
+  async function poll() {
+    try {
+      if (disablePolling()) return
+      await pollingCallback()
+      setTimeout(poll, intervalMs)
+    } catch (e) {
```

Comment:
I worry about wrapping the whole block inside a `try/catch` statement because many things can fail... and would be trace back to this (arguably the `pollingCallback` can trickle down to many things)

Reply:
And if any error happens deep in the stack, the `setTimeout` function will not be called anymore which means that polling will be halted forever.